app/src/main/java/org/oppia/android/app/mydownloads/downloads/DownloadsFragmentPresenter.kt

```diff
@@ -0,0 +1,205 @@
+
+  private lateinit var sortByListIndexListener: SortByListIndexListener
+
+  @Inject
+  lateinit var downloadsViewModel: DownloadsViewModel
+
+  private lateinit var downloadsRecyclerViewAdapter: BindableAdapter<DownloadsItemViewModel>
+  private lateinit var binding: DownloadsFragmentBinding
+  private var previousSortTypeIndex: Int? = null
+
+  fun handleCreateView(
+    inflater: LayoutInflater,
+    container: ViewGroup?,
+    internalProfileId: Int,
+    previousSortTypeIndex: Int?,
+    sortByListIndexListener: SortByListIndexListener
+  ): View? {
+    binding = DownloadsFragmentBinding.inflate(
+      inflater,
+      container,
+      /* attachToRoot= */ false
+    )
+    this.previousSortTypeIndex = previousSortTypeIndex
+    this.internalProfileId = internalProfileId
+    this.sortByListIndexListener = sortByListIndexListener
+
+    binding.apply {
+      this.lifecycleOwner = fragment
+      this.viewModel = downloadsViewModel
+    }
+
+    downloadsViewModel.setInternalProfileId(internalProfileId)
+    downloadsViewModel.setProfileId(internalProfileId)
+
+    binding.downloadsRecyclerView.apply {
+      downloadsRecyclerViewAdapter = createRecyclerViewAdapter()
+      adapter = downloadsRecyclerViewAdapter
+    }
+    return binding.root
+  }
+
+  private fun createRecyclerViewAdapter(): BindableAdapter<DownloadsItemViewModel> {
+    return BindableAdapter.MultiTypeBuilder
+      .newBuilder<DownloadsItemViewModel, ViewType> { downloadsItemViewModel ->
+        when (downloadsItemViewModel) {
+          is DownloadsSortByViewModel -> ViewType.VIEW_TYPE_SORT_BY
+          is DownloadsTopicViewModel -> ViewType.VIEW_TYPE_TOPIC
+          else -> throw IllegalArgumentException(
+            "Encountered unexpected view model: $downloadsItemViewModel"
+          )
+        }
+      }
+      .registerViewDataBinder(
+        viewType = ViewType.VIEW_TYPE_SORT_BY,
+        inflateDataBinding = DownloadsSortbyBinding::inflate,
+        setViewModel = this::bindDownloadsSortBy,
+        transformViewModel = { it as DownloadsSortByViewModel }
+      )
+      .registerViewDataBinder(
+        viewType = ViewType.VIEW_TYPE_TOPIC,
+        inflateDataBinding = DownloadsTopicCardBinding::inflate,
+        setViewModel = this::bindDownloadsTopicCard,
+        transformViewModel = { it as DownloadsTopicViewModel }
+      )
+      .build()
+  }
+
+  private enum class ViewType {
+    VIEW_TYPE_SORT_BY,
+    VIEW_TYPE_TOPIC
+  }
+
+  private fun bindDownloadsSortBy(
+    binding: DownloadsSortbyBinding,
+    viewModel: DownloadsSortByViewModel
+  ) {
+    binding.viewModel = viewModel
+    val adapter = ArrayAdapter(
+      fragment.requireContext(),
+      R.layout.downloads_sortby_menu,
+      SortByItems.values()
+    )
+    // setting input type to zero makes AutoCompleteTextView no editable
+    binding.sortByMenu.setInputType(0)
+    binding.sortByMenu.setText(adapter.getItem(previousSortTypeIndex ?: 0).toString(), false)
```

Comment:
Add inline comment for `false`. Here and everywhere else in this file

Reply:
done

Reply:
resolving as this is nit fix